base/server/cmscore/src/com/netscape/cmscore/dbs/DBVirtualList.java

```diff
@@ -437,6 +438,11 @@ private synchronized boolean getEntries() {
             String ldapAttrs[] = null;
             LDAPSearchResults result;
 
+            Integer size_limit = (Integer) mConn.getOption(LDAPv2.SIZELIMIT);
```

Comment:
This is the one change I'm not sure about. The Factory sets the `SIZELIMIT` to the maximum allowed; do we care about pruning it to a smaller amount when `mPageSize` is smaller? I think the answer here is yes, but I'm not sure. In the case that say, `SIZELIMIT = 10000`, but `mPageSize = 1`, this could be a considerable speed difference. But when the difference is small, I'm not sure it matters. Thoughts?

Reply:
I think in general the server should try to accommodate the client's parameters as long as it doesn't exceed the server's limit. So pruning is correct, I think.